includes/delete_datastream.form.inc

```diff
@@ -19,44 +19,15 @@
  *   The drupal form definition.
  */
 function islandora_delete_datastream_form(array $form, array &$form_state, AbstractDatastream $datastream) {
-  // XXX: Stashed version of datastream is deprecated... Use object and
-  // datastream IDs from 'datastream_info' to acquire.
   $form_state['datastream'] = $datastream;
-
-  $form_state['datastream_info'] = array(
-    'object_id' => $datastream->parent->id,
-    'datastream_id' => $datastream->id,
-  );
   $object = $datastream->parent;
   $dsid = $datastream->id;
-  $dsids = array_merge(array($dsid), islandora_datastream_to_purge($object, $dsid));
-  $dsids = array_unique($dsids);
+  $derivs = implode(', ', islandora_datastream_to_purge($object, $dsid));
   $form['delete_derivatives'] = array(
-    '#title' => t('Delete Derviatives'),
+    '#title' => t('Delete Derivatives'),
     '#type' => 'checkbox',
     '#default_value' => 0,
-    '#description' => t('Derivatives can be regenerated at a later time.'),
-  );
-  $form['base_info'] = array(
-    '#type' => 'item',
-    '#title' => t('Datastream to be purged'),
-    '#markup' => $dsid,
-    '#states' => array(
-      'invisible' => array(
-        ':input[name="delete_derivatives"]' => array('checked' => TRUE),
-      ),
-    ),
-  );
-  $form['derivative_info'] = array(
-    '#type' => 'item',
-    '#title' => t('Datastream(s) to be purged'),
-    '#description' => t('Including detectable derivatives.'),
-    '#markup' => implode(', ', $dsids),
-    '#states' => array(
-      'visible' => array(
-        ':input[name="delete_derivatives"]' => array('checked' => TRUE),
-      ),
-    ),
```

Comment:
Actually... Didn't notice this yesterday... Looks like you might be undoing some other work?